test/test_common/utility.h

```diff
@@ -369,6 +371,27 @@ class TestAllocator : public RawStatDataAllocator {
 private:
   static void freeAdapter(RawStatData* data) { ::free(data); }
   std::unordered_map<std::string, CSmartPtr<RawStatData, freeAdapter>> stats_;
+  const StatsOptions& stats_options_;
+};
+
+class MockedTestAllocator : public RawStatDataAllocator {
+public:
+  MockedTestAllocator(const StatsOptions& stats_options) : alloc_(stats_options) {
```

Comment:
Nit: constructor/destructor for mocks should live in `.cc`. See https://github.com/google/googlemock/blob/master/googlemock/docs/v1_6/CookBook.md#making-the-compilation-faster

Reply:
done